src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>(todos);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [option, setOption] = useState(0);
+  const [userError, setUserError] = useState(false);
+
+  const pushUser = (todo: Todo) => {
+    const newTodo = {
+      ...todo,
+    };
+
+    setCurrentTodos(currentTodo => [...currentTodo, newTodo]);
+  };
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTitle(value);
+    setTitleError(false);
+  };
+
+  const handleUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = e.target;
+
+    setOption(+value);
+    setUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setOption(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (option === 0) {
+      setUserError(true);
+    }
+
+    if (title && option !== 0) {
+      pushUser({
+        id: Math.max(...todos.map(({ id }) => id)) + 1,
+        title,
+        completed: false,
+        userId: 0,
+        user: getUser(option),
+      });
+
+      reset();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:&nbsp;
+            <input
+              type="text"
+              data-cy="titleInput"
+              value={title}
+              placeholder="Enter a title"
+              onChange={handleTitleChange}
+            />
+            {titleError && <span className="error">Please enter a title</span>}
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            User:&nbsp;
+            <select
+              data-cy="userSelect"
+              value={option}
+              onChange={handleUserChange}
+            >
+              <option value="0" disabled>Choose a user</option>
+              {usersFromServer.map((user) => (
+                <option value={user.id}>
```

Comment:
Since we have access to a unique value (`user.id`) it would be good to set it as key